From 9e05097d17d60e3ea0473d4befeef0d24a75167d Mon Sep 17 00:00:00 2001
From: Konstantin Porotchkin <kostap@marvell.com>
Date: Mon, 8 Feb 2021 05:06:59 -0800
Subject: [PATCH 1246/1921] arch/arm64: dts: fix the SD CD signal in CN9130-CRB

The CN9130-CRB uses SD card detect signal wired to CP0 MPP55.
Fix the CP0 SDHCI GPIOs list and add cd-gpios entry for proper
handling of this signal.

Signed-off-by: Konstantin Porotchkin <kostap@marvell.com>
Change-Id: I2f31cff4c62ba4f0e10064b58098e6598b1aa7ea
Reviewed-on: https://sj1git1.cavium.com/c/IP/SW/kernel/linux/+/45545
Tested-by: sa_ip-sw-jenkins <sa_ip-sw-jenkins@marvell.com>
Reviewed-by: Stefan Chulski <stefanc@marvell.com>
[WK: The original patch got from Marvell sdk11.21.09]
Signed-off-by: Wenlin Kang <wenlin.kang@windriver.com>
---
 arch/arm64/boot/dts/marvell/cn9130-crb.dts | 5 +++--
 1 file changed, 3 insertions(+), 2 deletions(-)

diff --git a/arch/arm64/boot/dts/marvell/cn9130-crb.dts b/arch/arm64/boot/dts/marvell/cn9130-crb.dts
index 22e04a2f3629..97f56316356b 100644
--- a/arch/arm64/boot/dts/marvell/cn9130-crb.dts
+++ b/arch/arm64/boot/dts/marvell/cn9130-crb.dts
@@ -186,8 +186,9 @@
 	status = "okay";
 	pinctrl-names = "default";
 	pinctrl-0 = <&cp0_sdhci_pins
-		     &cp0_sdhci_cd_pins>;
+		     &cp0_sdhci_cd_pins_crb>;
 	bus-width = <4>;
+	cd-gpios = <&cp0_gpio2 23 GPIO_ACTIVE_HIGH>;
 	vqmmc-supply = <&cp0_reg_sd_vccq>;
 	vmmc-supply = <&cp0_reg_sd_vcc>;
 };
@@ -263,7 +264,7 @@
 			marvell,pins = "mpp13", "mpp14", "mpp15", "mpp16";
 			marvell,function = "spi1";
 		};
-		cp0-sdhci-cd-pins-crb {
+		cp0_sdhci_cd_pins_crb: cp0-sdhci-cd-pins-crb {
 			marvell,pins = "mpp55";
 			marvell,function = "gpio";
 		};
-- 
2.31.1

